src/library/Select/Select.js

```diff
@@ -186,110 +230,171 @@ export default class Select extends Component<Props, State> {
 
   render() {
     const {
-      data,
       disabled,
       modifiers,
-      name,
-      placeholder,
       readOnly,
-      size,
-      variant,
-      triggerRef,
+      renderTrigger,
       ...restProps
     } = this.props;
     const isOpen = this.getControllableValue('isOpen');
-    const selectedItem = this.getControllableValue('selectedItem');
 
     const rootProps = {
-      id: this.id,
       ...restProps,
-      data,
+      id: this.id,
       disabled: disabled || readOnly,
       highlightedIndex: this.getHighlightedOrSelectedIndex(),
-      getMenuProps: this.getMenuProps,
-      getItemProps: this.getItemProps,
-      getTriggerProps: this.getTriggerProps,
       isOpen,
       modifiers: {
         contentWidth: contentWidthModifier,
         ...modifiers
       },
       onClose: this.close,
-      onOpen: this.open
+      onOpen: this.open,
+      renderMenu: this.renderMenu,
+      renderTrigger: renderTrigger ? this.renderTrigger : undefined
```

Comment:
Compare this to [Dropdown's rootProps](https://github.com/mineral-ui/mineral-ui/pull/666/commits/a6d756695e4a6ff5593f754fca9cdf326067ef2f#diff-1c3e1a871da8e1de1f623b5fa6b6af2eR167). Can we make them consistent?